core/src/main/java/org/infinispan/container/entries/ClusteredRepeatableReadEntry.java

```diff
@@ -82,7 +79,7 @@ private EntryVersion getCurrentEntryVersion(DataContainer container, Persistence
       EntryVersion prevVersion;// on origin, the version seen is acquired without the lock, so we have to retrieve it again
       // TODO: persistence should be more orthogonal to any entry type - this should be handled in interceptor
       InternalCacheEntry ice = PersistenceUtil.loadAndStoreInDataContainer(container, persistenceManager, getKey(),
-            ctx, timeService, ignored);
+            ctx, timeService, null);
```

Comment:
I'm not sure about this. wouldn't be cheaper to have the ```AtomicReference``` (```volatile``` write) instead of changing the ```loadAndStoreInDataContainer``` to check for ```null```?